src/net_processing.cpp

```diff
@@ -610,6 +625,42 @@ struct Peer {
             uint32_t short_txid = 1 + (s & 0xFFFFFFFF);
             return short_txid;
         }
+
+        /**
+         * Estimate a capacity of a sketch we will send or use locally (to find set difference)
+         * based on the local set size.
+         */
+        uint16_t EstimateSketchCapacity() {
+            uint16_t set_size_diff = std::abs(uint16_t(m_local_set.size()) - m_remote_set_size);
+            uint16_t min_size = std::min(uint16_t(m_local_set.size()), m_remote_set_size);
+            uint16_t weighted_min_size = m_remote_q * min_size;
+            uint16_t estimated_diff = 1 + weighted_min_size + set_size_diff;
+            return minisketch_compute_capacity(RECON_FIELD_SIZE, estimated_diff, RECON_FALSE_POSITIVE_COEF);
+        }
+
+        /**
+         * Reconciliation involves computing a space-efficient representation of transaction identifiers (a sketch).
+         * A sketch has a capacity meaning it allows reconciling at most a certain number of elements. (see BIP-330).
+         * Considering whether we are going to send a sketch to a peer or use locally, we estimate the set difference.
+         */
+        minisketch* ComputeSketch(const std::set<uint256> local_set, uint16_t& capacity)
+        {
+            std::vector<uint32_t> short_ids;
+            for (uint256 wtxid: local_set) {
+                uint32_t short_txid = ComputeShortID(wtxid);
+                short_ids.push_back(short_txid);
+                m_local_short_id_mapping.insert(std::pair<uint32_t, uint256>(short_txid, wtxid));
```

Comment:
Should the `m_local_short_id_mapping` be cleared out in this function?

Reply:
Why so early? The point of it is to refer to it later, when we are requested something by short ID.
That's why it should be kept until `FinalizeReconciliation`. 

Although as currently implemented, I notice there's a bug, because we may lose short IDs of the transactions received during extension when calling this after the extension is over. I need to fix this part.